src/main/java/org/cactoos/io/TailInput.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * <p>There is no thread-safety guarantee.
+ * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+public final class TailInput implements Input {
+
+    /**
+     * Input to decorate.
+     */
+    private final Input input;
+
+    /**
+     * Number of last bytes to show from the input.
+     */
+    private final int count;
+
+    /**
+     * Maximum number of bytes to read at once.
+     */
+    private final int max;
+
+    /**
+     * Constructor.
+     * @param inpt Input to decorate
+     * @param bytes Number of last bytes to show from input
+     */
+    public TailInput(final Input inpt, final int bytes) {
+        // @checkstyle MagicNumber (1 line)
+        this(inpt, bytes, 16384);
+    }
+
+    /**
+     * Constructor.
+     * @param inpt Input to decorate
+     * @param bytes Number of last bytes to show from input
+     * @param maximum Maximum number of bytes to read at once
+     */
+    public TailInput(final Input inpt, final int bytes, final int maximum) {
+        this.input = inpt;
+        this.count = bytes;
+        this.max = maximum;
+    }
+
+    @Override
+    public InputStream stream() throws IOException {
+        if (this.max < this.count) {
+            throw new IllegalArgumentException(
+                new FormattedText(
+                    "Can't tail %d bytes if buffer is set to %d",
+                    this.count, this.max
+                ).asString()
+            );
+        }
+        final byte[] buffer = new byte[this.max];
+        final byte[] response = new byte[this.count];
+        int num = 0;
+        final InputStream stream = this.input.stream();
+        while (true) {
+            final int read = stream.read(buffer);
+            if (read <= 0) {
+                break;
+            }
+            if (read < this.max && read < this.count) {
+                num = this.copy(buffer, response, num, read);
+                break;
+            }
+            System.arraycopy(
+                buffer, read - this.count, response, 0, this.count
+            );
+            num = Math.min(this.count, read);
+            if (read < this.max) {
+                break;
+            }
+        }
+        return new ByteArrayInputStream(response, 0, num);
+    }
+
+    /**
+     * Copy buffer to response for read count smaller then buffer size.
+     * @param buffer The buffer array
+     * @param response The response array
+     * @param num Number of bytes in response array from previous read
+     * @param read Number of bytes read in the buffer
+     * @return New count of bytes in the response array
+     * @checkstyle ParameterNumberCheck (3 lines)
+     */
+    private int copy(final byte[] buffer, final byte[] response, final int num,
+        final int read) {
+        final int result;
+        if (num > 0) {
+            System.arraycopy(
+                Arrays.copyOfRange(response, read, this.count),
```

Comment:
@krzyk  For the first look you can remove `Arrays.copyOfRange`. Like `System.arraycopy(response, read, response, 0, this.count-read)`